modules/dnn/src/net_impl.cpp

```diff
@@ -2068,6 +2069,39 @@ void Net::Impl::getMemoryConsumption(
     }
 }
 
+void Net::Impl::enableHighPrecision(bool highPrecision_)
+{
+    if (highPrecision != highPrecision_)
+    {
+        highPrecision = highPrecision_;
+
+        for (MapIdToLayerData::const_iterator it = layers.begin(); it != layers.end(); it++)
+        {
+            int lid = it->first;
+            LayerData &ld = layers[lid];
+            Ptr<Layer>& currLayer = ld.layerInstance;
+
+            if (ld.type == "Convolution")
+            {
+                Ptr<ConvolutionLayer> convLayer = ld.layerInstance.dynamicCast<ConvolutionLayer>();
+                if (!convLayer.empty())
+                    convLayer->highPrecision = highPrecision_;
+                else
+                    ld.params.set("high_precision", highPrecision_);
+            }
+
+            if (ld.type == "ConvolutionInt8")
```

Comment:
Sequence of `if` switches is anti-pattern.

API for layers should be propagated instead.

---

In the future we should separate:
- "imported" DL network - we could work with parameters.
- "execution" DL network - layers are instantiated for specified backend/device and optimization parameters. After that, parameters are locked.

Otherwise we would got a mess.

Reply:
I changed the API to `enableWinograd`, and it will only work on the `Convolution` and `ConvolutionInt8` layer. And this issue should go away.